bundler/lib/bundler/cli/install.rb

```diff
@@ -88,13 +88,6 @@ def run
         Bundler.ui.warn "Some gems seem to be missing from your #{Bundler.settings.app_cache_path} directory."
       end
 
-      unless Bundler.definition.has_rubygems_remotes?
-        Bundler.ui.warn <<-WARN, :wrap => true
-          Your Gemfile has no gem server sources. If you need gems that are \
-          not already on your machine, add a line like this to your Gemfile:
-          source 'https://rubygems.org'
-        WARN
-      end
```

Comment:
Something I also noticed is that rescuing `VersionConflict` above was probably only done to show this error message, so since we are removing it, we can now limit the rescue to the `GemNotFound` class.